docfx_project/articles/bootstrapping-trust/part-02.a-primer-on-public-key-cryptography.md

```diff
@@ -0,0 +1,168 @@
+and handwritten signatures. That puts handwritten signatures at a
+significant disadvantage. So how does this work?
+
+The data that will be signed can be anything. What it represents is not
+important. It can be text, JSON, an image, a PDF, or anything at all!
+That data processed by a one-way hashing algorithm, such as SHA-256.
+This process is idempotent, meaning running it repeatedly on the same
+data results and results in the same output. The output of this process
+is a hash, a string of characters that uniquely identifies the input
+data. With sufficiently large input data, the hash is much shorter than
+the input data as the hash size is usually fixed length.
+
+For example, here is the Ziti logo:
+
+![Image of a simple mesh](./images/ziti-system.png)
+
+This logo's file can be hashed using SHA-256 via the `sha256sum` command
+commonly found on Linux.
+
+```
+$> sha256sum ziti.png
+c3a6681cc81f9c0fa44b3e2921495882c55f0a86c54cd60ee0fdc7d200ad26db  ziti.png
+```
+
+That long string "c3a....6db" is the hash of that file! The string is 64
+characters long because it is a hex representation of the signature. The
+signature only needs 4 bits per character, so the result is: 64 x 4 =
+256, which is where SHA-256 gets its name.
+
+The hash itself is not encryption. It is "hashing." Hashing of this
+nature is not reversible, and it is impracticable to have two similar
+sets of data that have the same function that produces the same hash. In
+essence, the hash uniquely represents the data: all of it! Changing even
+a single character would generate a different hash.
+
+Next, the private key holder encrypts the hash with the private key to
+generate the signature. This process provides the following truths when
+working with the signature:
+
+- the private key is the only key capable of producing its signature of
+  the data
+- the public key can validate the signature given the data and hashing
+  algorithm used
+
+Verifying a signature a straightforward process:
+
+- Use the public key to decrypt the signature of the data to reveal the
+  original hash
+- Use the hashing algorithm that was used initially on the data,
+  recreate the hash independently
+- Compare the two hashes, and if they are the same the signature is
+  valid
+
+Signing data is incredibly powerful. It allows a private key holder to
+state that data was approved by them and not altered. It is also
+publicly verifiable. This allows many decentralized approaches to
+sharing data that can have its source and content verified.
+
+JSON Web Tokens (JWTs) are a prevalent form of bearer token. They are a
+document that is signed by a trusted authentication system and contain
+data that states the client's identity. This client can then present
+that JWT to any system which can then verify that the contents are valid
+and from a trusted identity provider as long as the public key is
+available. The bearer tokens can also contain claims which grant the
+bearer rights on a system.
+
```

Comment:
generally speaking at the end of 'parts' you get the "in part 'x' we will learn about blah-di-blah-di-blah" - might be nice to see that sort of thing at the end of parts 1-4

Reply:
addded